plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/math/KVector.java

```diff
@@ -435,10 +435,20 @@ public static double crossProduct(final KVector v, final KVector w) {
      * @return the rotated vector
      */
     public KVector rotate(final double angle) {
-        this.x = this.x * Math.cos(angle) - this.y * Math.sin(angle);
-        this.y = this.y * Math.sin(angle) + this.y * Math.cos(angle);
+        double newX = this.x * Math.cos(angle) - this.y * Math.sin(angle);
```

Comment:
Is rotate used by other methods that change their behavior once this works correctly?

Reply:
No, nothing is broken by this fix.